Graceful handling of KeyboardInterrupt when used with gunicorn

Hey there, great job on this WSGI framework, thanks so much for sharing it 😄 

I've discovered a minor issue when running Gunicorn with Meinheld workers in the foreground and hitting Ctrl+C to exit; it seems that KeyboardInterrupt isn't handled so we see an exception.

```
(venv) [centos@myserver my-benchmarks]$ gunicorn -w 1 --worker-class=meinheld.gmeinheld.MeinheldWorker api_falcon:app
[2018-05-16 12:39:07 +0000] [7341] [INFO] Starting gunicorn 19.8.1
[2018-05-16 12:39:07 +0000] [7341] [INFO] Listening at: http://127.0.0.1:8000 (7341)
[2018-05-16 12:39:07 +0000] [7341] [INFO] Using worker: meinheld.gmeinheld.MeinheldWorker
[2018-05-16 12:39:07 +0000] [7344] [INFO] Booting worker with pid: 7344
^C[2018-05-16 12:40:07 +0000] [7341] [INFO] Handling signal: int
[2018-05-16 12:40:07 +0000] [7344] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/home/centos/venv/lib64/python3.6/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "/home/centos/venv/lib64/python3.6/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.run()
  File "/home/centos/venv/lib64/python3.6/site-packages/meinheld/gmeinheld.py", line 46, in run
    server.run(self.wsgi)
KeyboardInterrupt
[2018-05-16 12:40:07 +0000] [7344] [INFO] Worker exiting (pid: 7344)
[2018-05-16 12:40:07 +0000] [7341] [INFO] Shutting down: Master
(venv) [centos@myserver my-benchmarks]$ 
```

I'm on Python 3.6.5 with the latest gunicorn and Meinheld from PyPI.

Here's my little `api_falcon.py` if it helps:

```python
import falcon


class HelloResource(object):
    def on_get(self, req, resp):
        resp.status = falcon.HTTP_200
        resp.body = 'Hello world!'


app = falcon.API()
app.add_route('/', HelloResource())
```

Any help would be greatly appreciated 😄 

Thanks
Fotis

--- a/meinheld/gmeinheld.py
+++ b/meinheld/gmeinheld.py
@@ -43,7 +43,10 @@ class MeinheldWorker(Worker):
         #server.set_watchdog(self.watchdog)
 
         server.set_listen_socket(fds)
-        server.run(self.wsgi)
+        try:
+            server.run(self.wsgi)
+        except KeyboardInterrupt:
+            pass
 
     def handle_quit(self, sig, frame):
         server.stop(int(self.timeout))

